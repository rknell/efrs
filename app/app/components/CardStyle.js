import React, { Component } from 'react'
import { Text, View } from 'react-native'
import { observer } from 'mobx-react/native'
import GlobalStyle from '../services/GlobalStyle'

@observer
class CardStyle extends Component {
  render () {
    return (
      <View style={[{ padding: 5, paddingBottom: 5 }, this.props.style]}>
        <View style={{
          backgroundColor: GlobalStyle.style.neutralColour,
          borderColor: GlobalStyle.style.neutralColourHighlight,
          flex: 1,
          borderRadius: 10,
          elevation: 1,
          overflow: 'hidden'
        }}>
          {!!this.props.title &&
          <View
            style={{ borderBottomColor: GlobalStyle.style.neutralColourHighlight, borderBottomWidth: 2 }}>
            <Text style={{ fontWeight: 'bold', fontSize: 18, color: GlobalStyle.style.neutralColourText, padding: 16 }}>
              {this.props.title} <Text
                style={{ fontWeight: 'normal', fontSize: 14, marginLeft: 16 }}>{this.props.subTitle}</Text>
            </Text>
          </View>
          }
          {this.props.children}
        </View>
      </View>
    )
  }
}

export default CardStyle
