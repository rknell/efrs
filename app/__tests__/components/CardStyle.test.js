import React from 'react'
import toJSON from 'enzyme-to-json'
import { shallow } from 'enzyme'
import CardStyle from '../../app/components/CardStyle'
import { Text } from 'react-native'

it('should render CardStyle', () => {
  const wrapper = shallow(<CardStyle><Text>Test</Text></CardStyle>)
  expect(toJSON(wrapper)).toMatchSnapshot()
  expect(wrapper.find(Text).length).toBe(1)
})

it('should render CardStyle with title', () => {
  const wrapper = shallow(<CardStyle title='Test title'><Text>Test</Text></CardStyle>)
  expect(toJSON(wrapper)).toMatchSnapshot()
  expect(wrapper.find(Text).length).toBe(3)
})
