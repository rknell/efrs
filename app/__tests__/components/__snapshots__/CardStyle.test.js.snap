// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CardStyle 1`] = `
<View
  style={
    Array [
      Object {
        "padding": 5,
        "paddingBottom": 5,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "borderColor": "rgb(217, 217, 217)",
        "borderRadius": 10,
        "elevation": 1,
        "flex": 1,
        "overflow": "hidden",
      }
    }
  >
    <Text>
      Test
    </Text>
  </View>
</View>
`;

exports[`should render CardStyle with title 1`] = `
<View
  style={
    Array [
      Object {
        "padding": 5,
        "paddingBottom": 5,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "borderColor": "rgb(217, 217, 217)",
        "borderRadius": 10,
        "elevation": 1,
        "flex": 1,
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "borderBottomColor": "rgb(217, 217, 217)",
          "borderBottomWidth": 2,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgb(51, 51, 51)",
            "fontSize": 18,
            "fontWeight": "bold",
            "padding": 16,
          }
        }
      >
        Test title
         
        <Text
          style={
            Object {
              "fontSize": 14,
              "fontWeight": "normal",
              "marginLeft": 16,
            }
          }
        />
      </Text>
    </View>
    <Text>
      Test
    </Text>
  </View>
</View>
`;
